### ISSUE TITLE
IN clause with tuple AND another expression match no rows

### ISSUE THREAD

**Issue by gleb-jiji** [yandex/ClickHouse#2146]
A query with a conjunction of an `IN` clause with a tuple and a basic equals operator doesn't match any rows it's expected to match.

```SQL
ClickHouse client version 1.1.54370.
Connecting to localhost:9000.
Connected to ClickHouse server version 1.1.54370.

:) CREATE TABLE foo
:-] (
:-]     date Date, 
:-]     bar String
:-] )
:-] ENGINE = MergeTree(date, (date), 8192);

CREATE TABLE foo
(
    date Date, 
    bar String
)
ENGINE = MergeTree(date, date, 8192)

Ok.

0 rows in set. Elapsed: 0.004 sec. 

:) INSERT INTO foo
:-] VALUES
:-] ( '2018-01-01', 'bar');

INSERT INTO foo VALUES

Ok.

1 rows in set. Elapsed: 0.004 sec. 
```
Unexpected behavior:
```SQL
:) SELECT *
:-] FROM foo 
:-] WHERE (date, bar) IN ('2018-01-01', 'bar') AND bar = 'bar';

SELECT *
FROM foo 
WHERE ((date, bar) IN ('2018-01-01', 'bar')) AND (bar = 'bar')

Ok.

0 rows in set. Elapsed: 0.003 sec.
```
Both conditions work as expected when used independently:
```SQL
:) SELECT *
:-] FROM foo 
:-] WHERE (date, bar) IN ('2018-01-01', 'bar');

SELECT *
FROM foo 
WHERE (date, bar) IN ('2018-01-01', 'bar')

┌───────date─┬─bar─┐
│ 2018-01-01 │ bar │
└────────────┴─────┘

1 rows in set. Elapsed: 0.004 sec. 

:) SELECT *
:-] FROM foo 
:-] WHERE bar = 'bar';

SELECT *
FROM foo 
WHERE bar = 'bar'

┌───────date─┬─bar─┐
│ 2018-01-01 │ bar │
└────────────┴─────┘

1 rows in set. Elapsed: 0.003 sec. 
```

**Comment by lamber-ken** [comment]
According to your steps，I test `ClickHouse server version 1.1.54343` and `ClickHouse server version 1.1.54343`

| version | result |  
| - |  -: |
| 1.1.54370 | wrong | 
| 1.1.54343 | right  |

**Comment by derekperkins** [comment]
I am running version 1.1.54370 and I am seeing a similar issue, even without a tuple.

`SELECT field1 FROM foo WHERE (dateTimeField IN '2018-03-02 00:00:00') AND (SomeID=1234);`
returns no rows

`SELECT field1 FROM foo WHERE (dateTimeField = '2018-03-02 00:00:00') AND (SomeID=1234);`
returns correct rows

This is a major data integrity issue.
cc @alex-zaitsev

**Comment by alex-zaitsev** [comment]
@derekperkins , 54370 has a number of problems with filters. These are regressions of some re-work performed by Yandex team in order to optimize in () queries that did not use indexes in some cases (e.g. in views). Some of those problems have been already addressed, so next release should be good. So far, the last release that I would recommend for prod use is 54343 -- it is ok except for some issues with UNION ALL.

**Comment by KochetovNicolai** [comment]
https://github.com/yandex/ClickHouse/pull/2166

**Comment by ludv1x** [comment]
It has been fixed in 1.1.54378.
